container/deployer.sh

```diff
@@ -0,0 +1,217 @@
+  container=$1
+  message=$2
+
+  if [ -n "$3" ]; then
+    count=$(($3))
+  else  
+    count=30
+  fi
+
+  while [ $count -gt 0 ]; do
+    if docker logs "$container" 2>&1 | grep "$message" >/dev/null; then
+      echo "OK - Container $container stand-up correctly - found in logs $message"
+      break
+    else
+      count=$((count-1))
+      sleep 1
+    fi    
+  done
+
+  if [ $count -eq 0 ]; then
+    echo "ERROR - Sorry the container $container didn't stand-up correctly"
+    exit 2
+  fi
+}
+
+function show_usage {
+  echo "Usage: $0 [options] <action>"
+  echo "Actions:"
+  echo " - prepare, prepare the environment. If uses --force the current directories will be removed and recreated"
+  echo " - db, create the database container. If uses --force the current db container will be removed and recreated"
+  echo " - webui, create the webui container. If uses --force the current configuration and webui container will be removed and ecreated"
+  echo " - worker, create the worker container If uses --force the current configuration and worker container will be removed and ecreated"
+  echo "Options:"
+  echo "-f|--force, destroys and recreate the previous data or container associated to the action"
+  echo "--prefix [network_name], set a prefix network and container creation (the default is openqa)"
+  echo "Notes:"
+  echo "  For initial deployments execute in order the actions: prepare, db, webui, worker"
+}
+
+function prepare {
+  [ $FORCE ] && rm -rf data 2>/dev/null
+  mkdir -p data/factory/{iso,hdd,other,tmp} data/{testresults,tests,conf} data/certs/{ssl.crt,ssl.key}
+  chmod a+w data/testresults
+  
+  [ $FORCE ] && docker network rm "$NETWORK" 2>/dev/null
+  docker network create "$NETWORK" 2>/dev/null
+  
+  echo "Next step (db), creation of the data base"
+}
+
+function db {
+  CONT_NAME="${PREFIX}_db"
+
+  [ $FORCE ] && docker rm -f "$CONT_NAME" 2>/dev/null
+  docker run -d --network "$NETWORK" -e POSTGRES_PASSWORD=openqa -e POSTGRES_USER=openqa -e POSTGRES_DB=openqa --net-alias=db --name "$CONT_NAME" postgres
+
+  wait_for_container_ready "$CONT_NAME"
+  wait_for_container_appropiate_logs "$CONT_NAME" "database system is ready to accept connections"
+
+  echo "Next step (webui), stand-up the webui"
+}
+
+function webui {
+  [ ! -d data ] && echo "Please execute first prepare command" && exit 1
+
+  CONT_NAME="${PREFIX}_webui"
+  IMAGE="registry.opensuse.org/devel/openqa/containers15.2/openqa_webui:latest"
+  
+  pushd .
+  cd data/conf || exit 1
+  wget https://raw.githubusercontent.com/os-autoinst/openQA/master/container/webui/conf/openqa.ini || exit 1
+  cp openqa.ini /tmp/
+  sed 's/method = OpenID/method = Fake/' </tmp/openqa.ini >openqa.ini
+  wget https://raw.githubusercontent.com/os-autoinst/openQA/master/container/webui/conf/database.ini || exit 1
+  wget https://raw.githubusercontent.com/os-autoinst/openQA/master/container/openqa_data/data.template/conf/client.conf || exit 1
```

Comment:
Now this gets tricky. So you want to mix the upstream container images with the current config files in git which might not be compatible.

Reply:
Maybe a solution could be to deliver this script with the configuration files. Therefore we ensure that won't be a discrepancy between the script and the config